tools/buildsteps/windows/vswhere.bat

```diff
@@ -0,0 +1,63 @@
+@ECHO OFF
+
+
+REM Trick to make the path absolute
+PUSHD %~dp0\..\..\..\project\BuildDependencies
+SET builddeps=%CD%
+POPD
+
+SET arch=%1
+SET vcarch=amd64
+SET vcstore=%2
+SET vcvars=no
+SET sdkver=
+
+SET vsver=
+SET toolsdir=%arch%
+
+IF "%arch%" NEQ "x64" (
+  SET vcarch=%vcarch%_%arch%
+)
+
+IF "%arch%"=="x86" (
+  SET toolsdir=win32
+)
+
+IF "%vcstore%"=="store" (
+  SET sdkver=10.0.14393.0
+  SET toolsdir="win10-%toolsdir%"
+)
+
+SET vswhere="%builddeps%\%toolsdir%\tools\vswhere\vswhere.exe"
+
+FOR /f "usebackq tokens=1* delims=" %%i in (`%vswhere% -latest -property installationPath`) do (
+  IF EXIST "%%i\VC\Auxiliary\Build\vcvarsall.bat" (
+    SET vcvars="%%i\VC\Auxiliary\Build\vcvarsall.bat"
+    SET vsver=15
+  )
+)
+
+IF %vcvars%==no (
+  FOR /f "usebackq tokens=1* delims=" %%i in (`%vswhere% -legacy -property installationPath`) do (
+    IF EXIST "%%i\VC\vcvarsall.bat" (
+      SET vcvars="%%i\VC\vcvarsall.bat"
+      SET vsver=14
```

Comment:
exit for loop after first existing path

Reply:
it's doesn't really matter much as for most systems we'll only get a single result back

Reply:
On my system I get two results back, although for the second one (12.0) the if check returns false but others may have installed older versions of VS.

Reply:
I was thinking of the wrong check. This check will be removed after we bump the sdk, it's only there currently to not break jenkins so we can do the upgrade in steps.

Also it seems harder than it should be to break out of a for loop see https://stackoverflow.com/questions/6728798/exiting-out-of-a-for-loop-in-a-batch-file haven't tested it myself though.